src/shogun/metric/LMNNImpl.cpp

```diff
@@ -131,49 +124,46 @@ SGMatrix<index_t> CLMNNImpl::find_target_nn(CDenseFeatures<float64_t>* x,
 	return target_neighbors;
 }
 
-MatrixXd CLMNNImpl::sum_outer_products(CDenseFeatures<float64_t>* x, const SGMatrix<index_t> target_nn)
+SGMatrix<float64_t> CLMNNImpl::sum_outer_products(
```

Comment:
I think this method could be pulled into linalg as it is useful for other situations as well.